### PR TITLE
Move matchers from cats into common, add smart constructors

### PR DESCRIPTION
**Pull request by dangerousben**
There's a lot of potential controversy here (doing it at all, using scalacheck, using cats-laws machinery in `common` etc) so I'm happy to do things differently (or not at all) if desired.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -186,4 +191,4 @@ lazy val root = (project in file("."))
   .settings(
     publish := {},
     publishLocal := {}
-  ) aggregate (core, scalatest, specs2, cats, scalaz)
+  ) aggregate (common, core, scalatest, specs2, cats, scalaz)
```
Comment (dangerousben) [inline build.sbt:194]: This is to make sure the tests from `common` run on `root/test`, there may be a better way.

**Comment by ultrasecreth** [comment]
So the dependencies are only for test code, and this would allow to do the same you did for cats in the scalaz module right? I don't see the drawback tbh (apart from the fact I'm gonna have to learn a bit more of scalacheck ;) )

**Comment by dangerousben** [comment]
> So the dependencies are only for test code

Yep.

> and this would allow to do the same you did for cats in the scalaz module right?

In theory yes, with the caveat that it's been a while since I've used scalaz.  I'm pretty sure it has the same or similar typeclasses, they just might be named differently.

> I don't see the drawback tbh (apart from the fact I'm gonna have to learn a bit more of scalacheck ;) )

Cool :)

**Comment by ultrasecreth** [comment]
Sorry for the delay, I thought I merged it :(

Will go in the next release :)

**Pull request merged**